### PR TITLE
dnssec clarifications

### PR DESCRIPTION
**Pull request by jsoref**
### Short description
minor fixes...

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: docs/dnssec/modes-of-operation.rst [hunk 1]
```diff
@@ -45,25 +45,25 @@ Zone Signing Keys (ZSKs). During normal operations, this means that only
...
-records, and lives with the DNSSEC keying material.
+As described above, there are several ways in which DNSSEC can deny the
+existence of a record, and this setting, which is also stored away from zone
+records, lives with the DNSSEC keying material.
 
 (Hashed) Denial of Existence
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-PowerDNS supports unhashed secure denial of existence using NSEC
+PowerDNS supports unhashed secure denial-of-existence using NSEC
```
Comment (Habbie) [inline docs/dnssec/modes-of-operation.rst]: why?

Comment (jsoref) [reply docs/dnssec/modes-of-operation.rst]: ## Parsing

### Short answer
Because it's hard to parse otherwise

### A portion of the community which writes it that way

https://hackerone.com/reports/169704 writes:
> NSEC and NSEC3 – For explicit denial-of-existence of a DNS record

https://www.coursehero.com/file/pr8555/DNSSEC-Lookup-67-Authenticated-Denial-of-Existence-Most-DNS-lookups-result-in/
> Authenticated Denial-of-Existence
> Most DNS lookups result in denial-of-existence NSEC (Next SECure)
> * Lists all extant RRs associated with an owner name
> * Easy zone enumeration

https://appsecco.com/books/subdomain-enumeration/active_techniques/zone_walking.html
> NSEC and NSEC3 | For explicit denial-of-existence of a DNS record

### See Also

It's similar to man-in-the-middle

### Technical Parsing

Probably the easiest way to understand it is to look at the [Stanford Parser](http://nlp.stanford.edu:8080/parser/index.jsp)'s parsing of:

> In addition, hashed secure denial of existence is supported using NSEC3 records, in two modes, one with help from the database, the other with the help of some additional calculations.

```
(ROOT
  (S
    (PP (IN In)
      (NP (NN addition)))
    (, ,)
    (NP
      (NP (JJ hashed) (JJ secure) (NN denial))
      (PP (IN of)
        (NP (NN existence))))
    (VP (VBZ is)
      (VP (VBN supported)
...
```

vs.

> In addition, hashed secure denial-of-existence is supported using NSEC3 records, in two modes, one with help from the database, the other with the help of some additional calculations.

```
(ROOT
  (S
    (PP (IN In)
      (NP (NN addition)))
    (, ,)
    (NP (JJ hashed) (JJ secure) (NN denial-of-existence))
    (VP (VBZ is)
      (VP (VBN supported)
...
```

## Alternatives

Note, the alternative (which is also used) would be `Denial of Existence` leading to `secure Denial of Existence`, but given the choice, I picked the hyphenated flavor over the capitalized flavor.

---
I was tempted to write something like: `PowerDNS supports unhashed, secure, denial of existence using NSEC`, but I can't figure out if `unhashed` and `secure` are _coordinate adjectives_ or _cumulative adjectives_. See [Grammar Girl: Commas with Adjectives](https://www.quickanddirtytips.com/education/grammar/commas-with-adjectives).

> In addition, hashed, secure, denial of existence is supported using NSEC3 records, in two modes, one with help from the database, the other with the help of some additional calculations.

```
(ROOT
  (S
    (PP (IN In)
      (NP
        (NP (NN addition))
        (, ,)
        (ADJP (JJ hashed) (, ,) (JJ secure))))
    (, ,)
    (NP
      (NP (NN denial))
      (PP (IN of)
        (NP (NN existence))))
    (VP (VBZ is)
      (VP (VBN supported)
...
```
